[netatmo] Binding does not reconnect after connection loss

This issues is discussed in the forum here : https://community.openhab.org/t/no-update-of-netatmo-items/57576/13

## Context
There has been a lot of connectivity errors on netatmo's side recently. 
These connectivity problems are logged as :
2018-11-22 21:49:37.404 [ERROR] [nternal.handler.NetatmoBridgeHandler] - Unable to connect Netatmo API : 502 Bad Gateway" in openhab logs.

The problem is that when the Netatmos server are back online, the binding does not update sensors values.

Just restarting the container does not restore the connection. Deleting and re-creating the container is necessary to have the binding work again.

## Your Environment
Openhab 2.4.0.M6 running in docker (amd64-debian)

--- a/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/NetatmoBridgeHandler.java
+++ b/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/NetatmoBridgeHandler.java
@@ -168,6 +168,10 @@ public class NetatmoBridgeHandler extends BaseBridgeHandler {
                             return;
                     }
                 }
+            } catch (Exception e) {
+                logger.warn("Unable to connect Netatmo API : {}", e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Netatmo Access Failed, will retry in " + configuration.reconnectInterval + " seconds.");
             }
             // We'll do this every x seconds to guaranty token refresh
         }, 2, configuration.reconnectInterval, TimeUnit.SECONDS);

--- a/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/NetatmoBridgeHandler.java
+++ b/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/NetatmoBridgeHandler.java
@@ -168,7 +168,7 @@ public class NetatmoBridgeHandler extends BaseBridgeHandler {
                             return;
                     }
                 }
-            } catch (Exception e) {
+            } catch (RuntimeException e) {
                 logger.warn("Unable to connect Netatmo API : {}", e.getMessage(), e);
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                         "Netatmo Access Failed, will retry in " + configuration.reconnectInterval + " seconds.");

